libs/langchain-community/src/vectorstores/azure_cosmosdb.ts

```diff
@@ -205,19 +241,24 @@ export class AzureCosmosDBVectorStore extends VectorStore {
 
   /**
    * Removes specified documents from the AzureCosmosDBVectorStore.
-   * @param ids IDs of the documents to be removed. If no IDs are specified,
-   *     all documents will be removed.
+   * If no IDs or filter are specified, all documents will be removed.
+   * @param params Parameters for the delete operation.
    * @returns A promise that resolves when the documents have been removed.
    */
-  async delete(ids?: string[]): Promise<void> {
+  async delete(params: AzureCosmosDBDeleteParams = {}): Promise<void> {
```

Comment:
This will be a breaking change. Instead we should overload this func so both input types are allowed, and we can narrow the type inside of the method.